Work in progress

davidlehn opened the pull request.

davidlehn committed:
Install local cuckoo lib.

davidlehn committed:
Major updates.

- Add rules to build blake2b lib.
- Add rules to build many solver libs.
- Load solver bindings based on params.
- Handle engine options.
- Update bindings for latest lib interface.
- Fix cuckoo lib branch.

davidlehn committed:
Add fixme.

davidlehn committed:
Add benchmark launchers.

dlongley commented on index.js:
```diff
@@ -3,17 +3,58 @@
 'use strict';
 
-const addon = require('bindings')('cuckoo');
+// FIXME: update to only load bindings on demand
```
+1

dlongley commented on index.js:
```diff
@@ -3,17 +3,58 @@
+
+  const addon = addons[`lean_${graphSize}_${edgeCount}`]
+  if(!addon) {
+    throw new RangeError(`Graph size of ${graphSize} not supported.`);
```
graph size AND edge count of X is not supported

davidlehn committed:
SipHash and parameter updates.

- Add SipHash type support.
- Update matrix of supported edgebits, proofsize, and siphash.
- Only load bindings on demand.

davidlehn committed:
Update dependencies.

davidlehn committed:
Support cuckatoo variant.

- Disable cuckoo support.
- Update built modules to make supporting multiple variants easier.
- Update binding.gyp to make supporting lean and mean miners easier.
- Use SQUASH_OUTPUT compile flag vs porting runtime debug flag patch.
- Update to use cuckatoo API.
- Use cuckoo libatoo branch.
